browser-extension/mv2/src/client/js/ruleExecutionHandler.js

```diff
@@ -141,3 +181,48 @@ RQ.RuleExecutionHandler.notifyRuleAppliedToWidget = (ruleId) => {
     );
   }
 };
+
+RQ.RuleExecutionHandler.notifyRuleAppliedToImplicitWidget = (rule) => {
+  const implicitTestRuleWidget = document.querySelector("rq-implicit-test-rule-widget");
+  implicitTestRuleWidget.style.display = "block";
+
+  if (implicitTestRuleWidget) {
+    implicitTestRuleWidget.setAttribute("applied-rule-id", rule.id);
+    implicitTestRuleWidget.setAttribute("applied-rule-name", rule.name);
+    implicitTestRuleWidget.setAttribute("applied-rule-type", rule.ruleType);
+    implicitTestRuleWidget.dispatchEvent(
+      new CustomEvent("new-rule-applied", {
+        detail: {
+          appliedRuleId: rule.id,
+          appliedRuleName: rule.name,
+          appliedRuleType: rule.ruleType,
+        },
+      })
+    );
+  }
+};
+
+const getImplicitTestRuleWidgetConfig = async () => {
```

Comment:
```suggestion
const RQ.RuleExecutionHandler.getImplicitTestRuleWidgetConfig = async () => {
```

Need to scope this inside a block otherwise global conflicts could happen in content script